### PR TITLE
update iter() for better fallback in getting 'meta' argument

### LINKED ISSUE #145: iter() not handling meta=None gracefully
I got a bit of a trouble when having the small code snippet below:

```
def get_spider_items(jobkey, meta=None):
    CLIENT.get_job(jobkey).items.iter(meta=meta)  # CLIENT was instantiated before

get_spider_items("1/2/3")
```

This results in something like:

```
  File "/app/python/lib/python3.6/site-packages/scrapinghub/client/proxy.py", line 113, in iter
    drop_key = '_key' not in apiparams.get('meta', [])
TypeError: argument of type 'NoneType' is not iterable
```

A workaround would something like the one below, although this would be much better when the package handles it:

```
meta = derive_something_with_meta() or []
get_spider_items("1/2/3", meta=meta)
```

### PR DESCRIPTION
**Pull request by BurnzZ**
Fixes #145 

Here's a condensed version of the code to verify the issue and fix:

```
def iter(**apiparams):
    print(apiparams)
    drop_key = '_key' not in apiparams.get('meta', [])
    print(drop_key)

def iter_fixed(**apiparams):
    print(apiparams)	
    drop_key = '_key' not in (apiparams.get('meta') or [])
    print(drop_key)


>>> iter(meta=None)
{'meta': None}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 3, in iter
TypeError: argument of type 'NoneType' is not iterable

>>> iter_fixed(meta=None)
{'meta': None}
True

>>> iter_fixed()
{}
True
```

### CODE REVIEW

**Review by vshlapakov — APPROVED** [review]
Accidentally spot an uncovered review request, sorry for that. It looks good, though I'm not a maintainer of the lib anymore, not sure if you need an approval from someone else on the subject.

**Comment by rafaelcapucho** [comment]
@vshlapakov Hey!, thank you Vik ✌

**Pull request merged**
